cmd/promtool/main.go

```diff
@@ -351,9 +397,9 @@ func CheckMetrics() int {
 }
 
 // QueryInstant performs an instant query against a Prometheus server.
-func QueryInstant(url *url.URL, query string) int {
+func QueryInstant(url string, query string) int {
 	config := api.Config{
-		Address: url.String(),
+		Address: url,
```

Comment:
Since you are introducing that new API HTTP client with built-in timeout in http.go, should it be used here and in the other function below as well?

Reply:
Actually I was thinking so, but I feel I should open another issue and resolve with another PR. Just easier for tracking the issues this way. What do you think?

Reply:
Yeah, since this PR is getting a bit busy, maybe let's do it in a separate PR.